### DISCUSSION TITLE
Audio glitch with Spotify

### DISCUSSION THREAD

**Discussion by fillods** [rern/rAudio#1198] [Bug Reports]
The Spotify renderer used to be broken with previous rAudio versions. After reconfiguring the Spotify renderer, and fixing the file `/etc/spotifyd.conf` which had duplicate `device/control/mixer` lines, it works fine (with a USB audio device) on the latest rAudio 20230117. 👍

However, there are audio glitches quite regularly, several times during a tune.

This is typical of system activity on a system with limited resources (BeagleBone Black, _single_ CPU).
A quick look at `top` on the system shows that the script `spotifyd.sh` is run with nice value **-19** (and sub-process like `curl`, `jq`, ..).
It makes sense that the `spotifyd` renderer runs with that priority, but not the helper script taking care of album cover, etc.

My quick workaround is to make the script `spotifyd.sh` to renice itself at startup.
```
--- a/srv/http/bash/spotifyd.sh
+++ b/srv/http/bash/spotifyd.sh
@@ -13,6 +13,7 @@
 . /srv/http/bash/common.sh
 
 ##### start
+renice 4 -p $$
 [[ $( < $dirshm/player ) != spotify ]] && $dirbash/cmd.sh playerstart$'\n'spotify && exit
 
 [[ $PLAYER_EVENT == volumeset ]] && $dirbash/cmd.sh volumepushstream
```

To whoever it may help.

**Comment by fillods** [comment]
Issue still present in 20230304.
I guess it affects any single CPU board.

**Comment by rern** [comment]
@fillods,

Please try set the niceness in `spotifyd.service` instead:
```sh
sed -i '/^RestartSec/ a\Nice=4' /etc/systemd/system/spotifyd.service
systemctl daemon-reload
systemctl restart spotifyd
```

Has anyone got the same issue on RPi?

**Reply by fillods** [reply]
@rern 

From what I tested, tweaking niceness in `spotifyd.service` does not help. Reading at sources, it can be explained by the `renice -n -19` done afterwards in `cmd.sh`:867.
This -19 niceness is then inherited by the script `spotifyd.sh`, which is troublesome on low-end SBC.

Ideally, the spotify daemon should take care by itself of the niceness in its player thread provided there's sufficient privileges (leaving along the `on_song_change_hook`). But I have no idea of what is done.

Is there any other workaround besides calling `renice` at beginning of `spotifyd.sh` ?
